src/Acme.UI/Infrastructure/Routing/AcmeContentFinder.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Acme.UI.Helper.Extensions;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Web;
+using Umbraco.Core;
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Web;
+using Umbraco.Web.Routing;
+
+namespace Acme.UI.Infrastructure.Routing
+{
+    public class AcmeContentFinder : IContentFinder
+    {
+        private readonly Dictionary<string, RouteParameters> _pathToContentType = new Dictionary<string, RouteParameters>
+        {
+            { "(?<home>dashboard)/(?<section>athlete)/(?<athleteName>[a-zA-Z0-9-]+)", new RouteParameters(typeof(Umbraco.Web.PublishedModels.Athlete), "Athlete", "//home/contentPage/contentPage")}
```

Comment:
Why did you need the regex in both the athletes controller and in this file?

Reply:
We probably need a simpler pattern in AcmeContentFinder. Here it is just trying to find the right matching record in _pathToContentType to assign the right published content (i.e. page) with the request. 

In Athlete controller we need value in the Url for athlete name to process it and get the athlete record.